Allow multiple incoming connections from the same IP

nbougalis opened the pull request:
Multiple servers behind NAT might share a single public IP, making it difficult for them to connect to the Ripple network since multiple incoming connections from the same non-private IP are currently not allowed.

RippleD can now automatically allow between 2 and 5 incoming connections, from the same public IP based on the total number of peers that it is configured to accept.

Administrators can manually change the limit by including an `[ip_limit]` stanza in their configuration file specifying a positive non-zero number. For example:

    [ip_limit]
    3

The server will automatically adjust this value to prevent a single IP from consuming all inbound slots.

The previous "one connection per IP" strategy can be emulated by simply setting `[ip_limit]` to 1 in the configuration file.

MarkusTeufelberger commented:
There's already [peers_max] - how about [peers_max_per_ip] or [peers_per_ip_max]?

nbougalis commented:
I'm not keen on `long_names_and_underscores` in the config file. I guess the name `ip_limit` is a bit non-descript. Anyone else have any feedback?

miguelportilla commented:
I suggest that you please update rippled-example.cfg file to reflect this new feature and its use.

nbougalis commented:
Pushed changes addressing  your comments. Thanks @miguelportilla!

miguelportilla commented:
:+1:

JoelKatz commented:
I probably would have added this as an overlay parameter, like `public_ip` is. But, either way, :+1:

nbougalis commented:
@JoelKatz: I have moved the configuration option to `[overlay]` instead of its own key. You can now do:

    [overlay]
    ip_limit = 3

I've rebased and squashed, and will leave the Passed label on this.
